test_suite/tests/derive.rs

```diff
@@ -142,6 +142,27 @@ fn c_like_enum_derive() {
     assert_type!(E, ty);
 }
 
+#[test]
+fn c_like_enum_derive_with_scale_index_set() {
+    #[allow(unused)]
+    #[derive(TypeInfo, Encode)]
+    enum E {
+        A,
+        B = 10,
+        #[codec(index = 13)]
+        C,
```

Comment:
I think we need two more cases to test:
-  `D` variant, ( == `3`?) to cover the case of a variant without an explicit index after some overridden ones.
-  `E` variant with both an explicit discriminant and an `index` attribute (index overrides discriminant?)

Reply:
> (index overrides discriminant?)

Yes: 1. `index` attr, 2. explicit discriminant, 3. order of appearance (aka implicit discriminant).

Will add more cases.